lib/mutant/ast/regexp/transformer.rb

```diff
@@ -0,0 +1,82 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        include AbstractType
+
+        REGISTRY = {} # rubocop:disable MutableConstant
```

Comment:
To be clear, I'm not talking about extracting `TypeRegistry` and having `Mutant::Mutator::Registry` inherit from it. It looks like this instead:

```ruby
module Mutant
  # Generic registry for mapping types
  class TypeRegistry
    include Concord.new(:contents)

    # Initialize object
    #
    # @return [undefined]
    def initialize
      super({})
    end

    # Raised when the type is an invalid type
    RegistryError = Class.new(TypeError)

    # Register entry class for AST node class
    #
    # @param [Symbol] type
    # @param [Class] entry
    #
    # @return [self]
    def add(type, entry)
      fail RegistryError, "Invalid type registration: #{type.inspect}" unless AST::Types::ALL.include?(type)
      fail RegistryError, "Duplicate type registration: #{type.inspect}" if contents.key?(type)
      contents[type] = entry
      self
    end

    # Lookup entry for node
    #
    # @param [Symbol] type
    #
    # @return [Class]
    #
    # @raise [RegistryError]
    #   raises error when entry is not found in registry
    def lookup(type)
      contents.fetch(type) do
        fail RegistryError, "No entry for: #{type.inspect}"
      end
    end

  end # Registry

end # Mutant
```

```ruby
module Mutant
  class Mutator
    # Registry for mutators
    class Registry
      include Concord.new(:type_registry)

      # Initialize object
      #
      # @return [undefined]
      def initialize
        super(TypeRegistry.new)
      end

      # Register mutator class for AST node class
      #
      # @param [Symbol] type
      # @param [Class:Mutator] mutator
      #
      # @return [self]
      def register(type, mutator)
        type_registry.add(type, mutator)
        self
      end

      # Call type_registry
      #
      # @return [Enumerable<Mutation>]
      def call(node, parent = nil)
        type_registry.lookup(node.type).call(node, parent)
      end

    end # Registry

    REGISTRY = Registry.new

  end # Mutator
end # Mutant
```